cirq-core/cirq/ops/pauli_string.py

```diff
@@ -1156,15 +1149,11 @@ def __mul__(self, other):
             return self._as_pauli_string() * other._as_pauli_string()
         if isinstance(other, (PauliString, numbers.Complex)):
             return self._as_pauli_string() * other
-        if (as_pauli_string := _try_interpret_as_pauli_string(other)) is not None:
```

Comment:
Actually can these whole dunder methods be removed? It inherits from PauliString, so I don't think they are needed. What happens if you remove them?

Reply:
Additionally I think the special case up in `PauliString.__mul__` for identity gate should be removable. Just trying to remove unnecessary code now as much as possible.